### PR TITLE
Add User Registration and Authentication

### LINKED ISSUE #81: Bridge User Creation
Bridge needs a resource for creating a brand new spanking user.

### LINKED ISSUE #102: Add a service for user registration and login
Bridge has no mechanism to communicate user actions.

### PR DESCRIPTION
**Pull request by benjic**
Problem
-------

Bridge has UI components for user registration and login but these do not function.

Solution
--------

Implement service to handle user registration and login functionality and wire it up to the UI components.

Howto Test
----------

- [ ] run `gulp`
- [ ] Create a user with the registration form
- [ ] Notice you are logged in with your new user
- [ ] Sign out
- [ ] Sign in with registration credentials
- [ ] Notice you are logged in again
- [ ] Logout
- [ ] Confirm no errors in console

References
----------

- @mkinsey This will require a bit of UI reworking to handle failures in
  registration/login. The modal should only close on good events.
- Resolves #81
- Closes #82 
- Resolves #102 
- Sorry to step on your toes @dill-wood :goberserk:

### CODE REVIEW

**Comment by benjic** [comment]
This PR uses a pending version of mission-control that faciliates user auth stuff. The tests require that you run a local version of mission-control. If anyone can make sure this works and merges the mission-control PR I will update this PR to point at the deployment. 

TLDR; DONT MERGE THIS UNTIL AFTER I UPDATE THE PATHS ONCE https://github.com/orbitable/mission-control/pull/52 IS MERGED.
